config.go

```diff
@@ -12,168 +12,200 @@
-		if err != nil {
-			return RecordConfig{}, fmt.Errorf("failed parsing fields: %w", err)
-		}
-		c.FormatType = formatType
-		c.FormatOptions = fields
-	default:
-		return RecordConfig{}, fmt.Errorf("unknown payload format %q", formatType)
-	}
+type ConfigBurst struct {
+	// The time the generator "sleeps" between bursts.
+	SleepTime time.Duration `json:"sleepTime"`
+	// The amount of time the generator is generating records in a burst. Has an
+	// effect only if `burst.sleepTime` is set.
+	GenerateTime time.Duration `json:"generateTime" default:"1s"`
+}
 
-	return c, nil
+type ConfigCollection struct {
+	// Comma separated list of record operations to generate. Allowed values are
+	// "create", "update", "delete", "snapshot".
+	Operation []string     `json:"operation" default:"create" validate:"required"`
+	Format    ConfigFormat `json:"format"`
 }
 
-type Config struct {
-	RecordCount  int64
-	ReadTime     time.Duration
-	SleepTime    time.Duration
-	GenerateTime time.Duration
-	RecordConfig RecordConfig
+type ConfigFormat struct {
+	// The format of the generated payload data (raw, structured, file).
+	Type string `json:"type" validate:"inclusion=raw|structured|file"`
+	// The options for the format type selected, which are:
+	//   1. For raw and structured: pairs of field names and field types, where the type can be one of: int, string, time, bool.
+	//   2. For the file format: a path to the file.
+	Options map[string]string `json:"options"`
+	// Path to the input file (only applicable if the format type is file).
+	FileOptionsPath string `json:"options.path"`
 }
 
-func Parse(config map[string]string) (Config, error) {
-	err := checkRequired(config)
-	if err != nil {
-		return Config{}, err
+func (c Config) Validate() error {
+	var errs []error
+
+	// Validate readTime and rate.
+	if c.ReadTime > 0 && c.Rate > 0 {
+		errs = append(errs, errors.New(`cannot specify both "readTime" and "rate", "readTime" is deprecated, please only specify "rate"`))
+	}
+	if c.ReadTime < 0 {
+		errs = append(errs, errors.New(`"readTime" should be greater or equal to 0`))
+	}
+	if c.Rate < 0 {
+		errs = append(errs, errors.New(`"rate" should be greater or equal to 0`))
 	}
 
-	dh := durationHelper{}
-	parsed := Config{}
+	// Validate burst.
+	if c.Burst.SleepTime < 0 {
+		errs = append(errs, errors.New(`"burst.sleepTime" should be greater or equal to 0`))
+	}
+	if c.Burst.SleepTime > 0 && c.Burst.GenerateTime <= 0 {
+		errs = append(errs, errors.New(`"burst.generateTime" should be greater than 0`))
+	}
 
-	// parse record count
-	// default value
-	parsed.RecordCount = -1
-	if recCount, ok := config[RecordCount]; ok {
-		recCountParsed, err := strconv.ParseInt(recCount, 10, 64)
+	// Validate collections.
+	collections := c.GetConfigCollections()
+	if len(collections) == 0 {
+		errs = append(errs, errors.New("invalid configuration, please configure at least one collection using `format.type` or `collections.*.format.type`"))
```

Comment:
I didn't remove the old options, in case we want to ever test what happens if the record doesn't contain a collection. IMO we don't necessarily need to deprecate them.